art/estimators/object_detection/tensorflow_faster_rcnn.py

```diff
@@ -0,0 +1,530 @@
+                      of (predictions, losses, detections):
+
+                        - predictions: a dictionary holding "raw" prediction tensors.
+                        - losses: a dictionary mapping loss keys (`Loss/RPNLoss/localization_loss`,
+                                  `Loss/RPNLoss/objectness_loss`, `Loss/BoxClassifierLoss/localization_loss`,
+                                  `Loss/BoxClassifierLoss/classification_loss`) to scalar tensors representing
+                                  corresponding loss values.
+                        - detections: a dictionary containing final detection results.
+        :param filename: Name of the file.
+        :param url: Download URL.
+        :param images: Input samples of shape (nb_samples, height, width, nb_channels).
+        :param sess: Computation session.
+        :param is_training: A boolean indicating whether the training version of the computation graph should be
+                            constructed.
+        :param clip_values: Tuple of the form `(min, max)` of floats or `np.ndarray` representing the minimum and
+                            maximum values allowed for input image features. If floats are provided, these will be
+                            used as the range of all features. If arrays are provided, each value will be considered
+                            the bound for a feature, thus the shape of clip values needs to match the total number
+                            of features.
+        :param channel_index: Index of the axis in data containing the color channels or features.
+        :param channels_first: Set channels first or last.
+        :param preprocessing_defences: Preprocessing defence(s) to be applied by the classifier.
+        :param postprocessing_defences: Postprocessing defence(s) to be applied by the classifier.
+        :param preprocessing: Tuple of the form `(subtractor, divider)` of floats or `np.ndarray` of values to be
+                              used for data preprocessing. The first value will be subtracted from the input. The
+                              input will then be divided by the second one.
+        :param attack_losses: Tuple of any combination of strings of the following loss components:
+                              `first_stage_localization_loss`, `first_stage_objectness_loss`,
+                              `second_stage_localization_loss`, `second_stage_classification_loss`.
+        """
+        # Remove in 1.5.0
+        if channel_index == 3:
+            channels_first = False
+        elif channel_index == 1:
+            channels_first = True
+        elif channel_index is not Deprecated:
+            raise ValueError("Not a proper channel_index. Use channels_first.")
+
+        # Super initialization
+        super().__init__(
+            clip_values=clip_values,
+            channel_index=channel_index,
```

Comment:
For new estimators we can ignore the channel_index argument and only use channels_first.
```suggestion
```

Reply:
Agree.